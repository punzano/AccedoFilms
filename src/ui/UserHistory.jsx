import React from 'react';
import { Link } from 'react-router-dom';

const UserHistory = (props) => {
    const { userHistoryList } = props;

    const getFormattedDate = (date) => new Date(date).toLocaleString();

    const getUserHistoryElements = (userHistoryList) => (
        userHistoryList.map(userHistoryEntry => (
            <li key={userHistoryEntry.video.id} className="history-entry-container" data-testid="history-entry">
                <span className="history-entry-date">{getFormattedDate(userHistoryEntry.date)}</span>
                <span className="history-entry-film-title">{userHistoryEntry.video.name}</span>
            </li>
        ))
    );

    return (
        <div className="user-history-container" data-testid="user-history-container">
            <header className="header">
                <h1 className="app-title">Accedo Films History</h1>
                <Link to="/">Back to home</Link>
            </header>
            <section className="history-container">
                {(userHistoryList && userHistoryList.length > 0) &&
                    <ul>
                        {userHistoryList.length > 0 && getUserHistoryElements(userHistoryList)}
                    </ul>
                }
                {(!userHistoryList || userHistoryList.length === 0) &&
                    <h1 data-testid="warning-history">¡Todavía no has visto ningún vídeo!</h1>
                }
            </section>
        </div>
    );
};

export default UserHistory;