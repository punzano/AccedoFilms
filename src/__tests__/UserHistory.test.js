import '@babel/polyfill';
import React from 'react';
import { render, fireEvent, cleanup, screen } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';
import UserHistory from '../ui/UserHistory';
import './mocks/localStorage';

afterEach(cleanup);

const mockPropsWithData = {
    userHistoryList: [
        {
            date: new Date("2022", "3", "7", "21", "00", "00"),
            video: {
                "id": 2259634,
                "name": "The Bump 139",
                "season": 4,
                "number": 5,
                "type": "regular",
                "image": "https://static.tvmaze.com/uploads/images/medium_portrait/217/542886.jpg",
                "mediaUrl": "https://test-data-interviews.s3.eu-west-1.amazonaws.com/bbb_multiaudio.mp4",
                "summary": "WWE's The Bump is unlike any WWE show you've ever seen before, featuring a dynamic cast of hosts--led by Kayla Braxton--and WWE Superstars stopping in weekly, both as in-studio and video call-in guests."
            
            }
        },
        {
            date: new Date("2022", "3", "7", "21", "30", "00"),
            video: {
                "id": 2262125,
                "name": "Lisa Ling, Amber Ruffin",
                "season": 1,
                "number": 2,
                "image": "https://static.tvmaze.com/uploads/images/medium_portrait/359/898286.jpg",
                "mediaUrl": "https://test-data-interviews.s3.eu-west-1.amazonaws.com/Forest+-+97998.mp4",
                "summary": "Lisa Ling guest co-hosts; TV host Amber Ruffin."
            }
        }
    ]
};

const mockPropsWithoutData = {
    userHistoryList: []
};

describe('UserHistory component', () => {
    test('UserHistory snapshot', async () => {
        render(
            <BrowserRouter>
                <UserHistory {...mockPropsWithData}/>
            </BrowserRouter>
        );
        const userHistoryContainer = screen.getByTestId('user-history-container');
        
        expect(userHistoryContainer).toMatchSnapshot();
    });

    test('UserHistory should render warning text if no history', async () => {
        render(
            <BrowserRouter>
                <UserHistory {...mockPropsWithoutData}/>
            </BrowserRouter>
        );
        
        const warningHistory = screen.getByTestId('warning-history');
        expect(warningHistory.textContent).toBe('¡Todavía no has visto ningún vídeo!');
    });

    test('UserHistory should render two entries in the history', async () => {
        render(
            <BrowserRouter>
                <UserHistory {...mockPropsWithData}/>
            </BrowserRouter>
        );
        
        const historyEntries = screen.getAllByTestId('history-entry');
        const title1 = historyEntries[0].querySelector('.history-entry-film-title').textContent;
        const title2 = historyEntries[1].querySelector('.history-entry-film-title').textContent;
        expect(historyEntries.length).toBe(2);
        expect(title1).toBe('The Bump 139');
        expect(title2).toBe('Lisa Ling, Amber Ruffin');
    });
});