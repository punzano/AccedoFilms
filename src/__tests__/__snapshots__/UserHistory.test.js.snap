// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHistory component UserHistory snapshot 1`] = `
<div
  class="user-history-container"
  data-testid="user-history-container"
>
  <header
    class="header"
  >
    <h1
      class="app-title"
    >
      Accedo Films History
    </h1>
    <a
      href="/"
    >
      Back to home
    </a>
  </header>
  <section
    class="history-container"
  >
    <ul>
      <li
        class="history-entry-container"
        data-testid="history-entry"
      >
        <span
          class="history-entry-date"
        >
          7/4/2022 21:00:00
        </span>
        <span
          class="history-entry-film-title"
        >
          The Bump 139
        </span>
      </li>
      <li
        class="history-entry-container"
        data-testid="history-entry"
      >
        <span
          class="history-entry-date"
        >
          7/4/2022 21:30:00
        </span>
        <span
          class="history-entry-film-title"
        >
          Lisa Ling, Amber Ruffin
        </span>
      </li>
    </ul>
  </section>
</div>
`;
